build/docker-entrypoint.sh

```diff
@@ -1,19 +1,27 @@
 #!/usr/bin/env bash
 
-if [ $1 == "dst-server" ]; then
-	set -e
+# Exit immediately on non-zero return codes.
+set -e
 
-	# Configure the server
+# Run boot scripts before starting the server.
+if [ $1 = 'dst-server' ]; then
+
+	# Prepare the shard directory.
 	mkdir -p $CLUSTER_PATH/$SHARD_NAME
 	chown -R $STEAM_USER:$STEAM_USER $CLUSTER_PATH
 
-	if [ "$2" == "start" ]; then
+	# Create configuration files.
+	if [ "$2" = 'start' ]; then
 		$DSTA_HOME/boot/token.sh
 		$DSTA_HOME/boot/settings.sh
 		$DSTA_HOME/boot/lists.sh
 		$DSTA_HOME/boot/world.sh
 		$DSTA_HOME/boot/mods.sh
 	fi
+
+	# Run as user `steam` if the command is `dst-server`.
+	set -- gosu steam "$@"
```

Comment:
Seen this in [various](https://github.com/docker-library/logstash/blob/master/5.0/docker-entrypoint.sh) other entrypoint scripts. I guess the idea is to run the containers process (`dst-server`) via gosu, while the rest runs on root. This change probably provokes some other changes to how we use gosu in the other scripts. I'd love to have this here and remove as many gosu calls in the scripts as possible, so it gets a lot clearer what is happening. Please share your thoughts on this with me!